zion/src/main/scala/edu/uci/ics/cloudberry/zion/model/impl/SparkSQLGenerator.scala

```diff
@@ -0,0 +1,636 @@
+        s"""$fieldExpr!="${values(0)}""""
+      }
+      case Relation.contains => ???
+
+    }
+  }
+  protected def parseTextRelation(filter: FilterStatement, fieldExpr: String): String = {
+    val wordsArr = ArrayBuffer[String]()
+    filter.values.foreach(w => wordsArr += "%" + w + "%")
+    val sb = new StringBuilder
+    for (i <- 0 until (wordsArr.length - 1)){
+      sb.append(s"lower($fieldExpr) ${fullTextContains} '${wordsArr(i)}' and ")
+    }
+    sb.append(s"lower($fieldExpr) ${fullTextContains} '${wordsArr(wordsArr.length - 1)}'")
+    sb.toString()
+  }
+
+
+  protected def parseGeoCell(scale: Double, fieldExpr: String, dataType: DataType.Value): String = {
+    val origin = s"${createPoint}(0.0,0.0)"
+    s"${getPoints}(${spatialCell}(${fieldExpr}, $origin, ${1 / scale}, ${1 / scale}))[0]"
+  }
+
+  protected def parseAggregateFunc(aggr: AggregateStatement,
+                                   fieldExpr: String): String = {
+    def aggFuncExpr(aggFunc: String): String = {
+      if (aggr.field.name.equals("*")) {
+        s"$aggFunc($groupVar)"
+      } else {
+        s"$aggFunc( (select value $groupVar.$fieldExpr from $groupVar) )"
+      }
+    }
+
+    aggr.func match {
+      case topK: TopK => ???
+      case DistinctCount => ???
+      case _ => aggFuncExpr(getAggregateStr(aggr.func))
+    }
+  }
+
+
+  private def parseFilter(filters: Seq[FilterStatement], exprMap: Map[String, FieldExpr], unnestTestStrs: Seq[String], queryBuilder: StringBuilder): ParsedResult = {
+    if (filters.isEmpty && unnestTestStrs.isEmpty) {
+      ParsedResult(Seq.empty, exprMap)
+    } else {
+      val filterStrs = filters.map { filter =>
+        parseFilterRelation(filter, exprMap(filter.field.name).refExpr)
+      }
+      val filterStr = (unnestTestStrs ++ filterStrs).mkString("where ", " and ", "")
+      appendIfNotEmpty(queryBuilder, filterStr)
+
+      ParsedResult(Seq.empty, exprMap)
+    }
+  }
+
+  private def parseUnnest(unnest: Seq[UnnestStatement],
+                          exprMap: Map[String, FieldExpr], queryBuilder: StringBuilder): ParsedResult = {
+    val producedExprs = mutable.LinkedHashMap.newBuilder[String, FieldExpr]
+    val unnestTestStrs = new ListBuffer[String]
+    val unnestStr = unnest.zipWithIndex.map {
+      case (unnest, id) =>
+        val expr = exprMap(unnest.field.name)
+        //        val newExpr = s"${quote}hash$id$quote"
+        val newExpr = s"${quote}hash$quote"
+        producedExprs += (unnest.as.name -> FieldExpr(newExpr, newExpr))
+        if (unnest.field.isOptional) {
+          unnestTestStrs += s"${expr.refExpr} is not null"
+        }
+        s"lateral view explode(${expr.refExpr}) tab as $newExpr"
+    }.mkString("\n")
+    appendIfNotEmpty(queryBuilder, unnestStr)
+
+    ParsedResult(unnestTestStrs.toSeq, (producedExprs ++= exprMap).result().toMap)
+  }
+
+  protected def parseGroupByFunc(groupBy: ByStatement, fieldExpr: String): String = {
```

Comment:
You should cleaning the privilege modifier of methods. Either make all of them as private, or protected.